lib/transaction.js

```diff
@@ -200,6 +154,64 @@ class Transaction extends EventEmitter {
     return this;
   }
 
+  async _kickThingsOff(config, container) {
+    // FYI: This is temporarily stalling things so that the constructor
+    //      can finish initializing the Transaction.  Otherwise,
+    //      `this.previousSibling` will still be `undefined`.
+    await Promise.resolve();
```

Comment:
This looks hacky. We can't reshuffle things around so that this method is only invoked after constructor is done?

Reply:
FYI: There is another conversation thread discussing this a few lines above.

I agree that this is "hacky"; however, this PR didn't actually introduce the hack.  It was implicitly occurring in the previous implementation as well.  Specifically, it:

1. Obtained a `connection`.  (which stalled things)
2. Waited for `_this.previousSibling`.  (Which was now defined thanks to the stall)
3. Continued processing `connection`.

So, this new implementation is simply making the hack more apparent.

Reply:
Makes sense :)